assigner/baserepo.py

```diff
@@ -67,40 +66,56 @@ def from_url(cls, url, token):
                    namespace, name, token, url)
 
         logging.debug(json.dumps(self.info))
-        logging.debug("{} is valid.".format(self.name_with_namespace))
+        logging.debug("%s is valid.", self.name_with_namespace)
 
         return self
 
     @classmethod
-    def _cls_gl_get(cls, url_base, path, token, params={}):
+    def _cls_gl_get(cls, url_base, path, token, params=None):
         """Make a Gitlab GET request"""
+        if not params:
+            params = {}
+
         params.update({"private_token": token})
         url = urljoin(url_base, "/api/v4" + path)
         r = requests.get(url, params=params)
         r.raise_for_status()
         return r.json()
 
     @classmethod
-    def _cls_gl_post(cls, url_base, path, token, payload={}, params={}):
+    def _cls_gl_post(cls, url_base, path, token, payload=None, params=None):
         """Make a Gitlab POST request"""
+        if not params:
```

Comment:
I'd love a better way to do this. Per pylint, default parameters that are objects are shared across all invocations.

Reply:
That is true, although in this case it's not a problem and perhaps even cuts down on the amount of garbage allocated (although frankly I'm really not worried about that for Assigner :smile:).

The reason why it's not a problem is that this function doesn't carry state in the default `params` or `payload` objects -- the default `payload` object just stays empty, and the default `params` object just has the `private_token` key which is updated every time around. I suppose we could worry about `requests.post` modifying them but it doesn't seem to right now.

In short: I'm ok with telling `pylint` to ignore that error in these cases.

Reply:
Awesome, I will go back and update this.